service/controller/v3/resource/chart/desired_test.go

```diff
@@ -214,3 +349,78 @@ func Test_DesiredState(t *testing.T) {
 	}
 
 }
+
+func Test_union(t *testing.T) {
+	testCases := []struct {
+		name         string
+		inputA       map[string]interface{}
+		inputB       map[string]interface{}
+		expectedMap  map[string]interface{}
+		errorMatcher func(error) bool
+	}{
+		{
+			name: "case 0: both maps with exclusive entries",
+			inputA: map[string]interface{}{
+				"secret": "secret",
+			},
+			inputB: map[string]interface{}{
+				"config": "config",
+			},
+			expectedMap: map[string]interface{}{
+				"secret": "secret",
+				"config": "config",
+			},
+		},
+		{
+			name: "case 1: only the first input",
+			inputA: map[string]interface{}{
+				"secret": "secret",
+			},
+			expectedMap: map[string]interface{}{
```

Comment:
Lets please always init the whole structure. Took me a while to realize that `inputB` is `nil`. Be explicit please.

Reply:
Fixed.